### ISSUE TITLE
Question: How to connect to Qliksense Enterprise from the browser

### ISSUE THREAD

**Issue by gangataarun** [qlik-oss/enigma.js#891]
Hi,

I'm building a qliksense mashup using React. I can authenticate and connect to the SaaS version of qliksense and would now like to connect to the Enterprise version of qliksense. How can I authenticate and connect from the browser?  Would that be possible and how?

**Comment by countnazgul** [comment]
The general idea is quite similar to the SaaS.

* try to establish connection
* if unauthenticated - redirect the user to the login page
* qlik redirects you back to the web app with qlikTicket in the url
* open connection and pass the ticket as url param in the engine connection string

Just published a post about it and saw your question :)

https://sstoichev.eu/2021/09/09/qseow-authentication-svelte/

**Comment by jlmelb** [comment]
Hi @countnazgul 

Thanks to your post, we are getting one step closer to being able to connect to our Enterprise version of QS!  We had managed to follow the steps you had outlined above and successfully connected using enigma.js by redirecting the user to the QS login page. 

We had a quick question on whether it would be possible to also retrieve the X-Qlik-Session cookie to our web app when the user logins into Qlik Sense?  The purpose for this is so that we can also perform other actions like using the Capability API via the session cookie / ticket that we had derived from connecting via enigma.js.

Hope the above makes sense.

Cheers,
Joel

**Comment by jlmelb** [comment]
Thanks @countnazgul.  Using your code, we have managed to authenticate and use enigma to connect to one of our apps.  

Our issue now is that we need to close the enigma session and create a new session to access a second app. Is there a way to do this without having the user to re-authenticate via the login page?  We are thinking whether there is a session cookie that we can pull back but would love some guidance!

**Comment by countnazgul** [comment]
@jlmelb i havent tested it myself but ...

You can use the ticket to access some protected resource (like an image or css file). Once the request is completed Qlik will send the session cookie and the browser will set it. After that you can connect to the engine (without applying the ticket in the web socket url)

**Comment by jlmelb** [comment]
Thanks @countnazgul 

When we tried to use the ticket to access a resource for example the QS require.js or style sheets, we are greeted with a status code of 302 Authenticate at this location.  Would this because the ticket has already been consumed by enigma via the execution of enigma.create() when returning to the browser with the ticket?


Example URL for accessing protected resource: https://host/resources/autogenerated/qlik-styles.css?QlikTicket=TicketString

**Comment by countnazgul** [comment]
@jlmelb yes. once the ticket is consumed it can't be used again. So what you can try is:
* try to open enigma connection. if its successful then all good
* if not successful then redirect the user to login
* once redirected back (after successful authentication) use the ticket the grab some file (`qlik-styles.css` in your case)
* once/if this is done try and open the engima connection again

Just to mention again that i havent tried this myself. If i have the time will try it

**Comment by countnazgul** [comment]
@jlmelb just to confirm that the above approach is working fine. Here is a simple [Svelte repo](https://github.com/countnazgul/qseow-svelte-authentication-session) that demonstrate it
